DCOS/fluentd-testing/preprovision-agent-windows.ps1

```diff
@@ -0,0 +1,60 @@
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+$TD_AGENT = "td-agent-3.1.1-0-x64.msi"
+$url = "http://packages.treasuredata.com.s3.amazonaws.com/3/windows/$TD_AGENT"
+$MSI_PATH = Join-Path $here $TD_AGENT
+$fluentPath = "$env:SystemDrive\opt\td-agent\embedded\bin"
+$env:Path = "$fluentPath;" + $env:Path.Replace("$fluentPath;", "")
+
+function Get-Fluentd () {
+    Write-Host "Downloading fluentd/td-agent..."
+    $wc = New-Object System.Net.WebClient
+    $wc.DownloadFile($url, $MSI_PATH)
+}
+
+function Install-Fluentd () {
+    Write-Host "Installing fluentd..."
+    
+    [System.IO.FileSystemInfo] $file = Get-Item $MSI_PATH
+    [String] $DataStamp = get-date -Format yyyyMMddTHHmmss
+    [String] $logFile = '{0}-{1}.log' -f $file.FullName, $DataStamp
+    
+    $MSIArguments = @(
+        "/i"                         # Install
+        ('"{0}"' -f $file.fullname)  # MSI path
+        "/qn"                        # No UI
+        "/norestart"                 # No computer restart
+        "/L*v"                       # Log everything including verbose
+        $logFile                     # Log file name
+    )
+    
+    $process = Start-Process "msiexec.exe" -ArgumentList $MSIArguments -Wait -NoNewWindow -PassThru
+    if ($process.ExitCode -ne 0) {
+        throw "ERROR: Failed to install Fluentd. Error code: " + $process.ExitCode + " See details in log: $logfile"
+    }
+}
+
+function Register-FluentdService () {
+    Write-Host "Registering fluentd service..."
+    fluentd --reg-winsvc i
```

Comment:
Is `fluentd` added to the `$env:PATH` when executing this code ?
Also, this is a call to a binary from the disk, so we need to check `$LASTEXITCODE` for any non-zero exit code.

After `fluentd` call, please add this check:

```
if($LASTEXITCODE) {
    Throw "Failed to register fluentd Windows service"
}
```

Reply:
fluentd is added to the Path at the top. Added the exit code check

Reply:
I wasn't sure if that refreshes the Path for the current session as well. But if that's the case, we're good.